### PR TITLE
fixed parseURL and all tests passed for bug: Value of request.url when using .inject differs from a real server #3257

### PR DESCRIPTION
**Pull request by LobRockyl**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
   "dependencies": {
     "ajv": "^8.1.0",
     "cookie": "^0.4.0",
+    "fastify": "^3.20.2",
```
Comment (mcollina) [inline package.json:9]: Unfortunately you should not be using fastify here :(. See the other tests, you can do this with a plain Node.js server.

**Review by mcollina — COMMENTED** [review]
Good work!

**Comment by LobRockyl** [comment]
`

          test('value of request url when using inject should not differ', async (t) => {
          t.plan(1)
          const ap = fastify();
          ap.get("*", function (request, reply) {
            reply.send({ url: request.url });
          });
          const res = await ap.inject({
            method: "GET",
            url: "http://example.com:8080//foo",
          });
          let v=JSON.parse((res.body))  //added this Json parse
        
          t.equal(v.url,'//foo')
          
        })
`

The code above works fine but as you said I removed fastify and did it with express. Now express doesn't have function inject so:
`

            test('value of request url when using inject should not differ', async (t) => {
              t.plan(1)
            
              const app = express()
            
              app.get('*', (req, res) => {
                console.log(req.url)
                res.jsonp({url:req.url})
              })
            
              inject(app, { method: 'GET', url: 'http://example.com:8080//hello' }, (err, res) => {
                console.log(res)
                t.equal(res.body, '')
              })
            })

`

But that gives an error and somehow the res.body is empty. Do you have any idea?

File: package.json (hunk 1 above)
Comment (LobRockyl) [reply package.json:9]: `

          test('value of request url when using inject should not differ', async (t) => {
          t.plan(1)
          const ap = fastify();
          ap.get("*", function (request, reply) {
            reply.send({ url: request.url });
          });
          const res = await ap.inject({
            method: "GET",
            url: "http://example.com:8080//foo",
          });
          let v=JSON.parse((res.body))  //added this Json parse
        
          t.equal(v.url,'//foo')
          
        })
`

The code above works fine but as you said I removed fastify and did it with express. Now express doesn't have function inject so:
`

            test('value of request url when using inject should not differ', async (t) => {
              t.plan(1)
            
              const app = express()
            
              app.get('*', (req, res) => {
                console.log(req.url)
                res.jsonp({url:req.url})
              })
            
              inject(app, { method: 'GET', url: 'http://example.com:8080//hello' }, (err, res) => {
                console.log(res)
                t.equal(res.body, '')
              })
            })

`

But that gives an error and somehow the res.body is empty. Do you have any idea?

Comment (mcollina) [reply package.json:9]: Do not use express, just use http.createServer().

Comment (LobRockyl) [reply package.json]: Hey did that, it's working could you review

**Review by mcollina — APPROVED** [review]
lgtm

**Review by Eomm — APPROVED** [review]
Fixes https://github.com/fastify/fastify/issues/3257

**Pull request merged**